go/vt/mysqlctl/mysqld.go

```diff
@@ -460,6 +460,10 @@ func cleanupLockfile(socket string, ts string) error {
 		log.Errorf("%v: error parsing pid from lock file: %v", ts, err)
 		return err
 	}
+	if os.Getpid() == p {
+		log.Infof("%v: lock file at %s is ours, removing it", ts, lockPath)
+		return os.Remove(lockPath)
```

Comment:
Do we really want to do this? I'm curious how this happens, where we have the same PID we had before? Or am I missing something. That being said, I suppose it's fine since eventually the PID will get reused and could be reused by another mysqld process. In that case though I'm also not sure what "ours" really means in that context.

Reply:
@mattlord It can happen inside containers. Each container is unique and gets new PIDs so it's possible there to have PID reusage much more often.

Reply:
Yes, when using persistent storage for the lock file that lives across containers. Makes sense. Thanks!